src/table.js

```diff
@@ -538,15 +543,105 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  const stringValue = typeof value === "string" ? value.trim() : value;
+  switch (type) {
+    case "string":
+      return typeof value === "string"
+        ? stringValue
+        : value || value === 0
+        ? value.toString()
+        : null;
+    case "boolean":
+      return value === true || stringValue === "true"
+        ? true
+        : value === false || stringValue === "false"
+        ? false
+        : null;
```

Comment:
In the case of the input _not_ being a string or a boolean, such as a number, I expect similar behavior to JavaScript’s boolean coercion. So rather than `0` and `1` both being coerced to `null`, I expect `0` to be coerced to `false` and `1` to be coerced to `true`. I likewise expect us to preserve undefined rather than coercing it to null.

```suggestion
      if (typeof value === "string") {
        const trimValue = value.trim();
        return trimValue === "true" ? true : trimValue === "false" ? false : null;
      }
      return typeof value === "boolean" || value == null
        ? value
        : Boolean(value);
```

I guess it’s a little inconsistent to use JavaScript’s boolean coercion for non-strings, and true/false testing for strings… but I think that’s unavoidable given that we’re using strings to represent “untyped” data.